server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java

```diff
@@ -103,34 +93,29 @@ public GradleSourceSets getGradleSourceSets(URI projectUri) {
       // via a copy constructor and return as a POJO.
       return new DefaultGradleSourceSets(customModelBuilder.get());
     } catch (GradleConnectionException | IllegalStateException | IOException e) {
-      summary = e.getMessage();
+      String summary = e.getMessage();
       if (errorOut.size() > 0) {
-        summary += "\n" + errorOut.toString();
+        summary += "\n" + errorOut;
       }
-      statusCode = StatusCode.ERROR;
+      reporter.sendError(summary);
```

Comment:
do you think if it is necessary to include the failure summary to the message of the task finish event? (A convenient way to display the errors of a certain task to users.)

Though, from the client side, it's still possible to establish the relationship between the error message and the build task - by the task id.

Reply:
The `TaskFinishParams` will still contain `StatusCode#Error` but, as you say, the message doesn't help the user zoom in on the issue.  The full message does get sent back as a `LogMessage` with `LogMessageParams#getType == MessageType#ERROR` but a client might ignore that.  One option would be to use `onBuildShowMessage` but I think that would be quite intrusive to the user as they'd get it popping up every time they saved with errors.

I think (but I'm not certain) that Metals just goes off the [OnBuildPublishDiagnostics](https://build-server-protocol.github.io/docs/specification#onbuildpublishdiagnostics-notification) which currently aren't sent.

I have a small draft PR for producing diagnostics which I'll submit after you're happy with this one.  But it only works on Gradle >=8.7 after the Problems API was introduced.

I've pushed some more changes where I've added a new `testProject/fail-compilation` which, as named, fails compilation and I've added integrity tests for it so we can see the way errors come back through log and finish messages.

I'm not really sure how this project will be able to handle reporting errors/warnings for Gradle <8.7.  Intellij seems to parse the stdout for java errors but that doesn't work for other languages e.g. Scala as the format is different.  Do you want to get diagnostics working for older versions of Gradle?  I'd probably just try and encourage users to upgrade.  Do you know of other projects besides Metals that are looking to use this project as a BSP server?

Reply:
> One option would be to use onBuildShowMessage but I think that would be quite intrusive to the user as they'd get it popping up every time they saved with errors.

Agree. `onBuildShowMessage` will be annoying.

> Do you want to get diagnostics working for older versions of Gradle?

At least, adopting the Gradle new problem API is a right way to approach. As for the older project, I think there is no need to support diagnostics right now. Unless it's frequently asked by community. So I'm fine to only support diagnostics for Gradle 8.7+.

> Do you know of other projects besides Metals that are looking to use this project as a BSP server?

AFAIK, right now only vscode java and Metals use this build server.